mlflow/dspy/wrapper.py

```diff
@@ -0,0 +1,45 @@
+import numpy as np
+import pandas as pd
+
+from mlflow.exceptions import INVALID_PARAMETER_VALUE, MlflowException
+
+
+class DspyModelWrapper:
+    """MLflow PyFunc wrapper class for Dspy models.
+
+    This wrapper serves two purposes:
+        - It stores the Dspy model along with dspy global settings, which are required for seamless
+            saving and loading.
+        - It provides a `predict` method so that it can be loaded as an MLflow pyfunc, which is
+            used at serving time.
+    """
+
+    def __init__(self, model, dspy_settings, **kwargs):
+        super().__init__(**kwargs)
+        self.model = model
+        self.dspy_settings = dspy_settings
+
+    def predict(self, inputs, **kwargs):
```

Comment:
Where is the `params` interface? For pyfunc, we'll definitely want to conform to that interface instead of using `**kwargs`

Reply:
Can we also add type hints to these public APIs?

Reply:
gotcha, and sure!